plugins/LilyPad/Linux/Config.cpp

```diff
@@ -81,29 +81,31 @@ struct GeneralSettingsBool
 };
 
 // XXX: I try to remove only gui stuff
-void DeleteBinding(int port, int slot, Device *dev, Binding *b)
+void DeleteBinding(int port, int slot, int padtype, Device *dev, Binding *b)
 {
     fprintf(stderr, "delete binding %d:%d\n", port, slot);
-    Binding *bindings = dev->pads[port][slot].bindings;
+    int active_padtype = config.padConfigs[port][slot].type;
+    Binding *bindings = dev->pads[port][slot][padtype].bindings;
     int i = b - bindings;
-    memmove(bindings + i, bindings + i + 1, sizeof(Binding) * (dev->pads[port][slot].numBindings - i - 1));
-    dev->pads[port][slot].numBindings--;
+    if (padtype == active_padtype) // Only do the memmove when it is the active padtype
+        memmove(bindings + i, bindings + i + 1, sizeof(Binding) * (dev->pads[port][slot][padtype].numBindings - i - 1));
```

Comment:
This still needs fixed.

Apart from that it seems fine and I see no obvious errors (I haven't tested the code myself though).